app/assets/javascripts/school_administrator_dashboard/dashboard_components/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -90,27 +93,46 @@ class SchoolDisciplineDashboard extends React.Component {
 
   }
 
+  sortedClassrooms(chartKeys) {
+    if(this.state.selectedChart == 'classroom') { //should always be true when this method is called
```

Comment:
If this is always supposed to happen, could this method always call `this.getChartData('classroom')` and then remove the if/else and comments?

Reply:
@kevinrobinson I'm not wild about that because the chartkeys have to be the keys from the classroom and there's nothing to check that's the case. 

But looking at this again I'm not sure why I thought this check was necessary. This function isn't called unless the selectedChart is classroom in the switch statement above, so I'll just remove this.

Reply:
👍 sounds good!

Yeah, I see what you're saying that you can't just pass any set of `chartKeys` to this method in any component state.  One thing to clarify might be naming the arguments for these methods more explicitly (and then `sortChartKeys` does the work of taking generic `chartKeys` and calling the right method).  Or i you want a stricter way to enforce this, you can move these sort methods to functions at the bottom of the file, outside the component.  Then you need to explicitly pass arguments, which would also show that `sortedClassrooms` needs all the chart data as well, in a way that the others don't.